### PR TITLE
Setup app page navigation

### PR DESCRIPTION
**Pull request by ismail-syed**
Closes https://github.com/vikrambombhi/SimpleSurvey/issues/47

This PR adds: 
* `react-router-dom` to handle app navigations
* Adds pages to the app and navigations between these pages 

## Home

Screen Shot 2020-03-18 at 2 47 06 AM

### A list of all the surveys for users to complete 

Screen Shot 2020-03-18 at 2 47 21 AM

### Similar to above but with admin privileges to create more surveys

Screen Shot 2020-03-18 at 2 46 48 AM

Next steps: 
- List the questions in the survey on the cards https://github.com/vikrambombhi/SimpleSurvey/issues/52
- `Complete Survey` link on cards should link to a page where users can complete surveys https://github.com/vikrambombhi/SimpleSurvey/issues/45
- `Edit Survey` link on admin cards should link to a page where uses can add questions to the survey

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "@testing-library/user-event": "^7.1.2",
     "react": "^16.13.0",
     "react-dom": "^16.13.0",
+    "react-router-dom": "^5.1.2",
```
Comment (dipshit) [inline client/package.json]: Are you able to fix the yarn warnings easily? I recognize that these are not from this PR though.

```
warning " > @testing-library/user-event@7.2.1" has unmet peer dependency "@testing-library/dom@>=5".
warning "react-scripts > eslint-config-react-app@5.2.0" has incorrect peer dependency "eslint-plugin-flowtype@3.x".
warning "react-scripts > @typescript-eslint/eslint-plugin > tsutils@3.17.1" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta".
```

File: src/main/java/com/example/survey/models/SurveyController.java [hunk 2]
```diff
@@ -13,7 +13,6 @@ public SurveyController(SurveyRepo surveyRepo) {
         this.surveyRepo = surveyRepo;
     }
 
-    @CrossOrigin(origins = "http://localhost:3000")
```
Comment (dipshit) [inline src/main/java/com/example/survey/models/SurveyController.java]: So it is the same origin if it has a different port?

File: client/yarn.lock [hunk 3]
```diff
@@ -5551,6 +5568,11 @@ is-wsl@^2.1.1:
   resolved "https://registry.yarnpkg.com/is-wsl/-/is-wsl-2.1.1.tgz#4a1c152d429df3d441669498e2486d3596ebaf1d"
   integrity sha512-umZHcSrwlDHo2TGMXv0DZ8dIUGunZ2Iv68YZnrmCiBPkZ4aaOhtv7pXJKeki9k3qJ3RJr0cDyitcl5wEH3AYog==
 
+isarray@0.0.1:
```
Comment (dipshit) [inline client/yarn.lock]: lmfao

**Comment by dipshit** [comment]
Looks great! I can't even describe how awesome this is as I am limited to prodeng jargon

File: client/package.json (hunk 1 above)
Comment (dipshit) [reply client/package.json]: The reason I care is that it hides test output when running `mvn test` from the intellij plugin. Maybe there is an alternate fix, like hiding yarn errors from mvn...

File: client/src/components/NewSurveyModal.js [hunk 4]
```diff
@@ -1,22 +1,53 @@
...
+
+  const handleModalAction = () => {
+    if (name === "") {
+      openHandler();
+      return;
+    }
+
+    async function submit() {
+      const res = await fetch(`/new?name=${name}`, {
+        method: "POST"
```
Comment (JackyChiu) [inline client/src/components/NewSurveyModal.js]: I guess spring would auto handle this with @RequestParam but shouldn't params be provided in the body for post requests?

File: client/package.json (hunk 1 above)
Comment (ismail-syed) [reply client/package.json]: Those yarn install warning really only show up on the first install. Subsequent `mvn tests` won't show those warnings.

File: src/main/java/com/example/survey/models/SurveyController.java (hunk 2 above)
Comment (ismail-syed) [reply src/main/java/com/example/survey/models/SurveyController.java]: This isn't needed. I forgot to remove it in my initial PR when I was trying to get CORS requests working. 

Instead, we specify a [`proxy`](https://github.com/vikrambombhi/SimpleSurvey/blob/master/client/package.json#L14) in the react app and the react development server will handle the requests for us appropriately.

File: client/src/components/NewSurveyModal.js (hunk 4 above)
Comment (ismail-syed) [reply client/src/components/NewSurveyModal.js]: We don't have our [controller](https://github.com/vikrambombhi/SimpleSurvey/blob/master/src/main/java/com/example/survey/models/SurveyController.java#L33) set up correctly to accept It as a body - [context](https://stackoverflow.com/questions/57623466/spring-boot-rest-required-string-parameter-is-not-present).

Comment (bkzhang) [reply client/src/components/NewSurveyModal.js]: see #33

**Pull request merged**